### PR TITLE
[13.0][FIX] stock_picking_report_valued_sale_mrp: multipicking

### PR DESCRIPTION
**Pull request by chienandalu**
Forward of #172 

When the recordset comes from multiple picking we could get a wrong calculation of the mrp lines in the report when the conditions of same component for the same order line would apply. So for every picking we have to compute the kits valuations independently.

cc @Tecnativa TT34194 TT33043

ping @victoralmau @pedrobaeza

### CODE REVIEW

File: stock_picking_report_valued_sale_mrp/models/stock_move_line.py [hunk 1]
```diff
@@ -37,10 +37,18 @@ def _compute_sale_order_line_fields(self):
...
+                lambda x: x.picking_id == picking
+            )._compute_sale_order_line_fields_by_picking()
+
+    def _compute_sale_order_line_fields_by_picking(self):
+        """We want to compute the lines value by picking to avoid mixing lines
+        if they wen't altogether"""
+        kit_lines = self.filtered("phantom_product_id")
         for sale_line in kit_lines.mapped("sale_line"):
             move_lines = kit_lines.filtered(lambda x: x.sale_line == sale_line)
             # Deduct the kit quantity from the first component in the picking.
```
Comment (pedrobaeza) [inline stock_picking_report_valued_sale_mrp/models/stock_move_line.py]:
```suggestion
            # Deduce the kit quantity from the first component in the picking.
```

File: stock_picking_report_valued_sale_mrp/models/stock_move_line.py [hunk 2]
```diff
@@ -37,10 +37,18 @@ def _compute_sale_order_line_fields(self):
...
             {"phantom_line": False, "phantom_delivered_qty": 0.0}
         )
+        for picking in pickings:
+            self.filtered(
+                lambda x: x.picking_id == picking
+            )._compute_sale_order_line_fields_by_picking()
+
+    def _compute_sale_order_line_fields_by_picking(self):
+        """We want to compute the lines value by picking to avoid mixing lines
+        if they wen't altogether"""
```
Comment (pedrobaeza) [inline stock_picking_report_valued_sale_mrp/models/stock_move_line.py]: Not sure if this is what you want to say:

```suggestion
        if they weren't shipped altogether.
        """
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I miss clear steps to reproduce the problem, for example: do this and that, and you get this problem/error, and if so, reflect such steps in a regression test.

**Comment by pedrobaeza** [comment]
@chienandalu can you please add the regression test?

**Comment by rousseldenis** [comment]
@chienandalu Could you attend @pedrobaeza comments ?

**Review by pedrobaeza — APPROVED** [review]
Let's merge this for not losing the fix:

/ocabot merge patch

**Pull request merged**